bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -179,17 +199,40 @@ export class AxisView extends GuideRendererView {
 
     const [nx, ny] = this.normals
     const standoff = extents.tick + extents.tick_label + this.model.axis_label_standoff
-
     const {vertical_align, align} = this.panel.get_label_text_heuristics("parallel")
-    axis_label.position = {
+    const angle = this.panel.get_label_angle_heuristic("parallel")
+
+    const position = {
       sx: sx + nx*standoff,
       sy: sy + ny*standoff,
       x_anchor: align,
       y_anchor: vertical_align,
     }
-    axis_label.align = align
 
-    axis_label.paint(ctx)
+    const axis_label = text instanceof MathText
+      ? this.axis_label_math_text_view
+      : new TextBox({text})
+
+    axis_label.angle = angle
+    axis_label.visuals = this.visuals.axis_label_text
+    axis_label.position = position
+
+    if (axis_label instanceof MathTextView) {
+      // If the image is not loaded
+      if (!axis_label.svg_image) {
+        // Request a new render
+        this.request_paint()
+        // Initiate loading if none has started yet
+        return axis_label.load_image()
+      }
+
+      // Once the image is finished loading draw it to Canvas
+      axis_label.draw_image(ctx)
+      this.notify_finished()
+    } else {
+      axis_label.align = align
+      axis_label.paint(ctx)
+    }
```

Comment:
This entire block will have to be collapsed into the original two lines, i.e.:
```
axis_label.align = align
axis_label.paint(ctx)
```
Accessing `notify_finished()` and `request_paint()` can be done view `this.parent` (assuming parent is `RendererView`, which it should be) in `MathTextView`.

Reply:
Why is
```ts
axis_label.align = align
```
only set in the second branch?

Reply:
We can't align the TeX/LaTeX rendered the same way we do with text, the user has to use respective TeX Macros http://docs.mathjax.org/en/latest/input/tex/macros/index.html?highlight=align